src/AspNetCoreModuleV2/AspNetCore/applicationmanager.cpp

```diff
@@ -148,101 +87,82 @@ APPLICATION_MANAGER::RecycleApplicationFromManager(
     _In_ LPCWSTR pszApplicationId
 )
 {
-    HRESULT                 hr = S_OK;
-    DWORD                   dwPreviousCounter = 0;
-    APPLICATION_INFO_HASH*  table = NULL;
-    CONFIG_CHANGE_CONTEXT   context;
-
-    if (g_fInShutdown)
+    try
     {
-        // We are already shutting down, ignore this event as a global configuration change event
-        // can occur after global stop listening for some reason.
-        return hr;
-    }
+        std::vector<std::shared_ptr<APPLICATION_INFO>> applicationsToRecycle;
 
-    {
-        SRWExclusiveLock lock(m_srwLock);
         if (g_fInShutdown)
         {
-            return hr;
+            // We are already shutting down, ignore this event as a global configuration change event
+            // can occur after global stop listening for some reason.
+            return S_OK;
         }
 
-        // Make a shallow copy of existing hashtable as we may need to remove nodes
-        // This will be used for finding differences in which applications are affected by a config change.
-        table = new APPLICATION_INFO_HASH();
-
-        // few application expected, small bucket size for hash table
-        if (FAILED(hr = table->Initialize(17 /*prime*/)))
         {
-            goto Finished;
+            SRWExclusiveLock lock(m_srwLock);
+            if (g_fInShutdown)
+            {
+                return S_OK;
+            }
+            const std::wstring configurationPath = pszApplicationId;
+
+            auto itr = m_pApplicationInfoHash.begin();
+            while (itr != m_pApplicationInfoHash.end())
+            {
+                if (itr->second->ConfigurationPathApplies(configurationPath))
+                {
+                    applicationsToRecycle.emplace_back(itr->second);
+                    itr = m_pApplicationInfoHash.erase(itr);
+                }
+                else
+                {
+                    ++itr;
+                }
+            }
+
+            // All applications were unloaded reset handler resolver validation logic
+            if (m_pApplicationInfoHash.empty())
+            {
+                m_handlerResolver.ResetHostingModel();
```

Comment:
Idea: Instead of resetting the handler resolver, we could just create it on creating the first application info and delete it when empty?

Reply:
I'll have to dynamically allocate it and add lazy loading logic. Not sure it's better.